### DISCUSSION TITLE
Using MSSQL with msnodesqlv8

### DISCUSSION THREAD

**Discussion by rbartholomay** [adonisjs/core#2713] [Help]
Hi there!

I have some problems with the latest version to connect to an SQL-Server. The Login will failed and it looks like a problem between knexjs / Tedious.

To test another driver i should replace 

```
require("mssql");
``` 

with 

```
require("mssql/msnodesqlv8");
``` 

How can i do this via adonisjs Database/Lucid?

Bye, René

**Comment by feniljariwala82** [comment]
Your solution lies in this documentation. Follow three steps if you are using Adonisjs v5 not legacy.
https://docs.adonisjs.com/guides/database/introduction 

Or if you are using the Legacy version then go through this link
https://legacy.adonisjs.com/docs/4.1/database

**Reply by rbartholomay** [reply]
thanks... i know how to connect to databases. But with the latest version i have problems to get the connection working and it seems a problem inside knex.

If i connect to the database via https://github.com/tediousjs/tedious all runs fine:

```javascript
  import { Connection, Request } from 'tedious';
  
  const config = {
    server: '192.168.1.50',
    authentication: {
      type: 'ntlm',
      options: {
        userName: '#########',
        password: '#########,
        domain: 'MyDomain'
      }
    },
    options: {
      port: 1433,
      encrypt: false,
      trustServerCertificate: true
    }
  };

  try {
      const connection = new Connection(config);
      
      connection.on('connect', (err) => {
        if (err) {
          console.log('Connection Failed');
          throw err;
        }

        console.warn("Connected")
      });


      connection.connect();

      return connection;

  } catch (error) {
    console.log(error)
    return error
  }
```


if i try the same via Adonis it fails...

**Reply by rbartholomay** [reply]
found it...

the authentication key cannot be used. All authentication key options must be inside the root like this:

```javascript
	{
      client: 'mssql',

      //@ts-ignore
      connection: {
        server: '192.168.1.50',
        port: 1433
        user: '#########',
        password: '#########,
        domain: 'MyDomain'


        // @ts-ignore
        type: 'ntlm',

        // @ts-ignore
        options: {
          enableArithAbort: false,
          encrypt: false,
        },
      },
      healthCheck: true,
    }
```

in my case it will run fine now...
